internal/pkg/deploy/upload/asset/asset.go

```diff
@@ -180,12 +183,12 @@ func (u *ArtifactBucketUploader) uploadAssetMappingFile(assets []asset) (string,
 
 // dedupe returns a copy of assets with duplicate entries removed.
 func dedupe(assets []asset) []asset {
-	type key struct{ a, b string }
+	type key struct{ a, b, c string }
```

Comment:
nit:
```go
type key struct{ field1, field2, field3 string }
```

Reply:
Added in https://github.com/aws/copilot-cli/pull/4870